deployment/data-streams/changeset/call_configurator_promote_staging_config.go

```diff
@@ -0,0 +1,193 @@
+package changeset
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+
+	owner_helpers "github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/mcms"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/deployment/common/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/llo-feeds/generated/configurator"
+)
+
+// PromoteStagingConfigConfig contains the parameters needed for each chain
+type PromoteStagingConfigConfig struct {
+	PromotionsByChain map[uint64][]PromoteStagingConfig
+	MCMSConfig        *MCMSConfig
+}
+
+type PromoteStagingConfig struct {
+	ConfiguratorAddress common.Address
+	// 32-byte configId
+	ConfigID [32]byte
+	// Whether the current production config is considered green
+	IsGreenProduction bool
+}
+
+type ConfiguratorState struct {
+	Configurator *configurator.Configurator
+}
+
+func (cfg PromoteStagingConfigConfig) Validate() error {
+	if len(cfg.PromotionsByChain) == 0 {
+		return errors.New("PromotionsByChain cannot be empty")
+	}
+	return nil
+}
+
+func CallPromoteStagingConfig(e deployment.Environment, cfg PromoteStagingConfigConfig) (deployment.ChangesetOutput, error) {
+	if err := cfg.Validate(); err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("invalid PromoteStagingConfigConfig: %w", err)
+	}
+
+	chainSelectors := make([]uint64, 0, len(cfg.PromotionsByChain))
+	for chainSelector := range cfg.PromotionsByChain {
+		chainSelectors = append(chainSelectors, chainSelector)
+	}
+
+	var (
+		mcmsStatePerChain map[uint64]*changeset.MCMSWithTimelockState
+		err               error
+	)
+	if cfg.MCMSConfig != nil {
+		mcmsStatePerChain, err = changeset.MaybeLoadMCMSWithTimelockState(e, chainSelectors)
+		if err != nil {
+			return deployment.ChangesetOutput{}, err
+		}
+	}
+
+	mcmsPerChain := map[uint64]*owner_helpers.ManyChainMultiSig{}
+	timelockAddresses := map[uint64]common.Address{}
+	allBatches := []timelock.BatchChainOperation{}
```

Comment:
I believe we should be using the types from [mcms lib](https://github.com/smartcontractkit/mcms/blob/92c4722f677c6eb7633a7b12b2cd02e34e8787b5/types/operation.go#L27) - these reference the older mcms lib